tokio/src/sync/watch.rs

```diff
@@ -458,23 +451,8 @@ mod state {
 /// [`Sender`]: struct@Sender
 /// [`Receiver`]: struct@Receiver
 pub fn channel<T>(init: T) -> (Sender<T>, Receiver<T>) {
-    let shared = Arc::new(Shared {
-        value: RwLock::new(init),
-        state: AtomicState::new(),
-        ref_count_rx: AtomicUsize::new(1),
-        notify_rx: big_notify::BigNotify::new(),
-        notify_tx: Notify::new(),
-    });
-
-    let tx = Sender {
-        shared: shared.clone(),
-    };
-
-    let rx = Receiver {
-        shared,
-        version: Version::initial(),
-    };
-
+    let tx = Sender::new(init);
+    let rx = tx.subscribe();
     (tx, rx)
```

Comment:
The `channel` method is used a lot more often than `Sender::new` will be, so this should be the version that is more efficient. If you want to define one in terms of the other, then you can define `new` by calling `channel`.

Reply:
ok I get it. I see 2 options about what to do in `Sender::new`

1. call the `channel` method and le the receiver be dropped
2. build the Shared struct as done in `channel`, "duplicating" that lines basically

Personally I would go with 1, but please let me know your opinion